dnf5/commands/clean/clean.cpp

```diff
@@ -131,6 +131,9 @@ void CleanCommand::run() {
     std::error_code ec;
     libdnf5::repo::RepoCache::RemoveStatistics statistics{};
     for (const auto & dir_entry : std::filesystem::directory_iterator(cachedir, ec)) {
+        if (!dir_entry.is_directory()) {
+            continue;
+        }
         libdnf5::repo::RepoCache cache(ctx.base.get_weak_ptr(), dir_entry.path());
 
         if (required_actions & CLEAN_ALL) {
```

Comment:
Should exceptions from these cache operations also be caught?

Reply:
The exception is caught, so dnf5 will not end up with core dump (all command.run() execution is inside a try/catch block). But you are right, it should be caught also here, otherwise the `clean` command execution will fail on the first inaccessible directory.